<h2>Why is this an issue?</h2>
<p>Private functions that are never executed are dead code: unnecessary, inoperative code that should be removed. Cleaning out dead code decreases the
size of the maintained codebase, making it easier to understand the program and preventing bugs from being introduced.</p>
<h3>Noncompliant code example</h3>
<pre>
public class Foo
{
  private function Foo(){}   //Compliant, private empty constructor intentionally used to prevent any direct instantiation of a class.
  public static function doSomething():void
  {
    var foo:Foo = new Foo();
    ...
  }
  private function unusedPrivateFunction():void {...}
}
</pre>
<h3>Compliant solution</h3>
<pre>
public class Foo
{
  private function Foo(){}   //Compliant, private empty constructor intentionally used to prevent any direct instantiation of a class.
  public static function doSomething():void
  {
    var foo:Foo = new Foo();
    ...
  }
}
</pre>

