<p>This rule raises an issue when a private function is never referenced in the code.</p>
<h2>Why is this an issue?</h2>
<p>A function that is never called is dead code, and should be removed. Cleaning out dead code decreases the size of the maintained codebase, making
it easier to understand the program and preventing bugs from being introduced.</p>
<p>This rule detects functions that are never referenced from inside a translation unit, and cannot be referenced from the outside.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
public class Foo
{
  public static function doSomething(): void  // Compliant - public function
  {
    var foo:Foo = new Foo();
    ...
  }

  private function unusedPrivateFunction(): void {...}  // Noncompliant
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
public class Foo
{
  public static function doSomething(): void  // Compliant - public function
  {
    var foo:Foo = new Foo();
    ...
  }
}
</pre>

