<h2>Why is this an issue?</h2>
<p>Try to imagine using the standard Flex API without ASDoc. It would be a nightmare, because ASDoc is the only way to understand of the contract of
the API.</p>
<p>Documenting an API with ASDoc increases the productivity of the developers use it.</p>
<h3>Noncompliant code example</h3>
<pre>
public class MyClass {
  public var myLabel:String;

  public function myMethod(param1:String):Boolean {...}
}
</pre>
<h3>Compliant solution</h3>
<pre>
/**
 * my doc
 */
public class MyClass {
  /**
   * my doc
   */
  public var myLabel:String;

  /**
   * my doc
   * @param param1 my doc
   * @return my doc
   */
  public function myMethod(param1:String):Boolean {...}
}
</pre>
<h3>Exceptions</h3>
<p>Classes or class elements with an ASDoc <code>@private</code> comment are ignored by this rule.</p>
<pre>
/**
 * @private  // This class and all its elements are ignored
 */
public class MyClass {  // Compliant

  public var myLabel:String;   // Compliant
}

public class AnotherClass {  // Noncompliant; class not @private and not documented

  /**
   * @private
   */
  public var name:String;  // Compliant
}
</pre>

