<p>A good API documentation is a key factor in the usability and success of a software API. It ensures that developers can effectively use, maintain,
and collaborate on the API.</p>
<h2>Why is this an issue?</h2>
<p>Undocumented APIs pose significant challenges in software development for several reasons:</p>
<ul>
  <li> <strong>Lack of Clarity:</strong> developers struggling to understand how to use the API correctly. This can lead to misuse and unexpected
  results. </li>
  <li> <strong>Increased Development Time:</strong> developers spending extra time reading and understanding the source code, which slows down the
  development process. </li>
  <li> <strong>Error Prone:</strong> developers are more likely to make mistakes that lead to bugs or system crashes when the intent or the error
  handling of an API is not clear. </li>
  <li> <strong>Difficult Maintenance and Updates:</strong> developers may not understand the existing functionality well enough to add new features
  without breaking the existing ones. </li>
  <li> <strong>Poor Collaboration:</strong> collaboration, when there is lack of documentation, leads to confusion and inconsistencies. </li>
</ul>
<p>It is recommended to document the API using <strong>ASDoc</strong> to clarify what is the contract of the API. This is especially important for
public APIs, as they are used by other developers.</p>
<h3>Exceptions</h3>
<p>Classes or class elements with an ASDoc <code>@private</code> comment are ignored by this rule.</p>
<pre>
/**
 * @private  // This class and all its elements are ignored
 */
public class MyClass {  // Compliant

  public var myLabel:String;   // Compliant
}

public class AnotherClass {  // Noncompliant; class not @private and not documented

  /**
   * @private
   */
  public var name:String;  // Compliant
}
</pre>
<h2>How to fix it</h2>
<p>Add the missing <strong>ASDoc</strong> for the public classes, methods, properties and metadata.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
public class MyClass {
  public var myLabel:String;

  public function myMethod(param1:String):Boolean {
    // ...
  }
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
/**
 * my doc
 */
public class MyClass {
  /**
   * my doc
   */
  public var myLabel:String;

  /**
   * my doc
   * @param param1 my doc
   * @return my doc
   */
  public function myMethod(param1:String):Boolean {
    // ...
  }
}
</pre>
<h2>Resources</h2>
<h3>Articles &amp; blog posts</h3>
<ul>
  <li> Technical Writer HQ - <a href="https://technicalwriterhq.com/documentation/api-documentation/how-to-write-api-documentation/">How to write API
  documentation</a> </li>
  <li> FreeCodeCamp - <a href="https://www.freecodecamp.org/news/how-to-write-api-documentation-like-a-pro/">How to write API documentation like a
  pro</a> </li>
</ul>

