<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>Making a public constant just <code>const</code> as opposed to <code>static const</code> leads to duplicating its value for every instance of the
class, uselessly increasing the amount of memory required to execute the application.</p>
<h3>Noncompliant code example</h3>
<pre>
public class Myclass
{
  public const THRESHOLD:int = 3;
}
</pre>
<h3>Compliant solution</h3>
<pre>
public class Myclass
{
  public static const THRESHOLD:int = 3;
}
</pre>

