<h2>Why is this an issue?</h2>
<p>Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.</p>
<h3>Noncompliant code example</h3>
<pre>
function doSomething(a:int, b:int):void      // "b" is unused
{
  compute(a);
}
</pre>
<h3>Compliant solution</h3>
<pre>
function doSomething(a:int):void
{
  compute(a);
}
</pre>
<h3>Exceptions</h3>
<p>The following cases are ignored</p>
<ul>
  <li> event handlers. </li>
  <li> overriding methods. </li>
  <li> all methods in classes implementing one or more interfaces. </li>
  <li> methods which are empty or where the body consists of a single comment or a single <code>throw</code> statement (i.e. where the intention is
  apparently to simulate an abstract class). </li>
</ul>
<pre>
override function doSomething(a:int):void {    // ignored
  compute(a);
}

...

class AbstractSomething {
  public function doSomething(a:int) {  // ignored
    throw new IllegalOperationError("doSomething() is abstract");
  }

...

interface I {
  function action(a:int, b:int);
}

class C extends I {
  function action(a:int, b:int) { // ignored
    return doSomethignWith(a);
  }
}

function clickHandler(event:MouseEvent):void { // ignored
   trace("click");
}
</pre>

