<h2>Why is this an issue?</h2>
<p>Shared naming conventions improve readability and allow teams to collaborate efficiently. This rule checks that all package names match a provided
regular expression.</p>
<h2>How to fix it</h2>
<p>Rename packages with the expected naming convention</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<p>With the default regular expression <code>^[a-z]+(\.[a-z][a-z0-9]*)*$</code>:</p>
<pre data-diff-id="1" data-diff-type="noncompliant">
package org.Example {  // Noncompliant
...
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
package org.example {
...
}
</pre>

