<h2>Why is this an issue?</h2>
<p>Even though this is syntactically correct, the <code>void</code> return type should not be used in the signature of a constructor. Indeed some
developers might be confused by this syntax, believing that the constructor is in fact a standard function.</p>
<h3>Noncompliant code example</h3>
<pre>
public class Foo
{
  public function Foo() : void
  {...}
}
</pre>
<h3>Compliant solution</h3>
<pre>
public class Foo
{
  public function Foo()
  {...}
}
</pre>

