<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>Declaring the package and class together has been deprecated since ActionScript 3. The package definition should be declared outside of the class
definition even if the old syntax is still supported.</p>
<h3>Noncompliant code example</h3>
<pre>
class P.A {...}
</pre>
<h3>Compliant solution</h3>
<pre>
package P {
    class A {...}
}
</pre>

