<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>Using several "--" or "++" unary operators in the same arithmetic expression can quickly make the expression unreadable.</p>
<h3>Noncompliant code example</h3>
<pre>
var j:int = foo++ - --bar;
</pre>
<h3>Compliant solution</h3>
<pre>
bar--;
var j:int = foo++ - bar;
</pre>

