<p>Using several "--" or "++" unary operators in the same arithmetic expression can quickly make the expression unreadable.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var j:int = foo++ - --bar;
</pre>
<h2>Compliant Solution</h2>
<pre>
bar--;
var j:int = foo++ - bar;
</pre>

