<p>Declaring the package and class together has been deprecated since ActionScript 3. The package definition should be declared outside of the class
definition even if the old syntax is still supported. </p>
<h2>Noncompliant Code Example</h2>
<pre>
class P.A {...}
</pre>
<h2>Compliant Solution</h2>
<pre>
package P {
    class A {...}
}
</pre>

