<h2>Why is this an issue?</h2>
<p>A listener can be attached to an object only after it has been constructed. So dispatching an event in a constructor is useless and error
prone.</p>
<h3>Noncompliant code example</h3>
<pre>
public class MyClass
{
  public function MyClass()
  {
    dispatchEvent( new Event( "uselessEvent" ) );
  }
}
</pre>

