<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>Calling Security.allowDomain("*") lets any domain cross-script into the domain of this SWF and exercise its functionality.</p>
<h3>Noncompliant code example</h3>
<pre>
Security.allowDomain("*");
</pre>
<h3>Compliant solution</h3>
<pre>
Security.allowDomain("www.myDomain.com");
</pre>

