<h2>Why is this an issue?</h2>
<p>Quoted from the Flex documentation :</p>
<blockquote>
  <p>When you define an Array variable in ActionScript, you specify Array as the data type of the variable. However, you cannot specify the data type
  of the elements of the Array.</p>
  <p>To allow the Flex MXML compiler to perform type checking on Array elements, you can use the [ArrayElementType] metadata tag to specify the
  allowed data type of the Array elements.</p>
</blockquote>
<h3>Noncompliant code example</h3>
<pre>
public var newStringProperty:Array;
public var newNumberProperty:Array;
</pre>
<h3>Compliant solution</h3>
<pre>
[ArrayElementType("String")]
public var newStringProperty:Array;

[ArrayElementType("Number")]
public var newNumberProperty:Array;
</pre>

