<h2>Why is this an issue?</h2>
<p>Access modifiers define which classes can access properties, variables, methods, and other classes. If an access modifier is not specified, the
access level defaults to <code>internal</code>, which grants access to all classes in the same package. This may be what is intended, but it should be
specified explicitly to avoid confusion.</p>
<p>Available access modifiers are:</p>
<ul>
  <li> <code>internal</code> - access allowed within the same package </li>
  <li> <code>private</code> - access allowed only within the same class </li>
  <li> <code>protected</code> - access allowed to the class and its child classes </li>
  <li> <code>public</code> - unfettered access by all </li>
</ul>
<h3>Noncompliant code example</h3>
<pre>
function checkResources():Boolean {
  ...
  return true;
}
</pre>
<h3>Compliant solution</h3>
<pre>
public function checkResources():Boolean {
  ...
  return true;
}
</pre>

