<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>It can be expensive to instantiate a new object, and doing so inside a loop is typically an error. Instead, create the object once, before the
loop.</p>
<h3>Noncompliant code example</h3>
<pre>
for (var i:int = 0; i &lt; 10; i++) {
  var temp:MyObj = new MyObject();  // Noncompliant
  //...
}
</pre>
<h3>Compliant solution</h3>
<pre>
var temp:MyObj = new MyObject();
for (var i:int = 0; i &lt; 10; i++) {
  //...
}
</pre>

