<h2>Why is this an issue?</h2>
<p>Having multiple cases in a <code>switch</code> with the same condition is confusing at best. At worst, it’s a bug that is likely to induce further
bugs as the code is maintained.</p>
<p>If the first case ends with a break, the second case will never be executed, rendering it dead code. Worse there is the risk in this situation that
future maintenance will be done on the dead case, rather than on the one that’s actually used.</p>
<p>On the other hand, if the first case does not end with a break, both cases will be executed, but future maintainers may not notice that.</p>
<h3>Noncompliant code example</h3>
<pre>
switch(i) {
  case 1:
    //...
    break;
  case 5:
    //...
    break;
  case 3:
    //...
    break;
  case 1:  // Noncompliant
    //...
    break;
}
</pre>
<h3>Compliant solution</h3>
<pre>
switch(i) {
  case 1:
    //...
    break;
  case 5:
    //...
    break;
  case 3:
    //...
    break;
}
</pre>

