<h2>Why is this an issue?</h2>
<p>The <code>onEnterFrame</code> event handler is continually invoked at the frame rate of the SWF file, regardless of which individual movie frame it
is set for. Having too many <code>onEnterFrame</code> handlers can seriously degrade performance.</p>
<p>If the use of this event handler cannot be avoided entirely, then it should be created as close to its use as possible, and then destroyed as soon
as possible afterward.</p>
<h3>Noncompliant code example</h3>
<pre>
movieClip.onEnterFrame = function () {   // Noncompliant
   // ...
}
</pre>

