<p>Never use <code>with</code> statements, since they decrease readability. When you do not specify a variable's scope, you do not always know where
you are setting properties, so your code can be confusing.</p>
<h2>Noncompliant Code Example</h2>
<pre>
with (foo) { // Noncompliant
  return x;  // is it a property of foo or local variable ?
}
</pre>
<h2>Compliant Solution</h2>
<pre>
return foo.x;
</pre>

